### ISSUE TITLE
SaveAsWidget broken in Firefox

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#3878]
@peller @billreed63 @jhpedemonte 

Not sure yet what's going wrong, but it appears that SaveAsWidget works with Chrome but not with Firefox.
- New desktop app
- Add a DIV
- Put LABEL and Button in the DIV
- Select the DIV
- SaveAsWidget

If created using FF, the resulting widget fails when trying to use it. It looks like the HTML file isn't getting written properly on Firefox, but I haven't researched this in depth yet.

**Comment by peller** [comment]
reopening until we work out the issues using Deferred directly from the setContents API

**Comment by peller** [comment]
still fails on FF.  HTML file is uploaded, but it shows up on server as foo.html.workingcopy instead of foo.html.  @JonFerraiolo had you seen this problem?

**Comment by JonFerraiolo** [comment]
> foo.html.workingcopy

Well, I'm not sure if I've seen this problem or not. You see, I did see one instance of this when I was working on a fix, but didn't see it after I finished my work. However, there is a clear suspect: the second parameter to setContents() is `isWorkingCopy`.

**Comment by peller** [comment]
I've seen it work intermittently on FF after my fix.  Yes, the second parameter to setContents is a flag, but the de code isn't setting it anywhere I can see.

**Comment by peller** [comment]
So stepping through these two lines in resource.js:

```
                        var html = widgetNamespace.createResource(widgetData.name + ".html");
                        promises.push(html.setContents(content.html));
```

neither one says anything about a workingCopy extension or flag on the client.  createResource makes a file with an .html.workingCopy extension on the server.  (note that this call uses sync xhrs) Then, the setContents either updates that file or renames it to .html, depending on whether I'm in the debugger or not.  setContents is async, but shouldn't start running before createResource finishes.  Some timing issue on the server?  @jhpedemonte would you happen to know anything about this code?
